lib/server/index.js

```diff
@@ -491,6 +495,46 @@ class Relay {
     })
   }
 
+  /**
+   * @param {string} id
+   * @param {string} directory
+   * @param {object} [options]
+   * @param {number} [options.limit]
+   * @param {number} [options.offset]
+   *
+   * @returns {lmdb.RangeIterable}
+   */
+  _searchDirectory (id, directory, options = {}) {
+    const range = { start: '/' + id + directory, end: '/' + id + directory.slice(0, directory.length - 1) + '0', ...options }
+
+    return this._recordsDB.getRange(range)
+  }
+
+  /**
+   * @param {http.IncomingMessage} _req
+   * @param {http.ServerResponse} res
+   * @param {URL} url
+   */
+  async _QUERY (_req, res, url) {
+    const [id, ...rest] = url.pathname.slice(1).split('/')
+    const dir = '/' + rest.join('/')
+
+    const options = {
+      // Skip options until we really need them. currently, it should be cheap to return 1000s of keys at once.
+
+      // limit: parseInt(url.searchParams.get('limit')) || 1000,
+      // offset: parseInt(url.searchParams.get('offset')) || 0
+    }
+
+    const keys = this._searchDirectory(id, dir, options).asArray.map(r => {
+      this._updateLastQueried(r.key)
+      return r.key
+    })
+
```

Comment:
```js
    const options = {
      // Skip options until we really need them. currently, it should be cheap to return 1000s of keys at once.

      // limit: parseInt(url.searchParams.get('limit')) || 1000,
      // offset: parseInt(url.searchParams.get('offset')) || 0

      startDate: parseInt(url.searchParams.get('startDate')) || 0,
      endDate: parseInt(url.searchParams.get('endDate')) || 0
    }

    const { limit, offset } = options
    const keys = this._searchDirectory(id, dir, { limit, offset }).asArray.map(r => {
      if (r.key.startsWith(SERVER_SIDE_RECORDS_METADATA)) return null

      const { startDate, endDate } = options
      if (startDate || endDate) {
        const deserialized = Record.deserialize(r.value)
        const timestamp = !deserialized.error ? deserialized.value.timestamp : -1
        if (timestamp < startDate || timestamp > (endDate || Date.now())) return null
      }

      this._updateLastQueried(r.key)
      return r.key
    }).filter(Boolean)
```
would allow query by record timestamp range

Reply:
I would like to keep queries as pure key-value store in the scope of datastore, queries like this make more sense in Backends for specific applications, or if you name your records using timestamps (or secondary indicies).